samtranslator/model/api/http_api_generator.py

```diff
@@ -87,6 +98,137 @@ def _construct_http_api(self):
 
         return http_api
 
+    def _construct_api_domain(self, http_api):
+        """
+        Constructs and returns the ApiGateway Domain and BasepathMapping
+        """
+        if self.domain is None:
+            return None, None, None
+
+        if self.domain.get("DomainName") is None or self.domain.get("CertificateArn") is None:
+            raise InvalidResourceException(
+                self.logical_id, "Custom Domains only works if both DomainName and CertificateArn" " are provided."
+            )
+
+        self.domain["ApiDomainName"] = "{}{}".format(
+            "ApiGatewayDomainNameV2", logical_id_generator.LogicalIdGenerator("", self.domain.get("DomainName")).gen()
+        )
+
+        domain = ApiGatewayV2DomainName(
+            self.domain.get("ApiDomainName"), attributes=self.passthrough_resource_attributes
+        )
+        domain_config = dict()
+        domain.DomainName = self.domain.get("DomainName")
+        endpoint = self.domain.get("EndpointConfiguration")
+
+        if endpoint is None:
+            endpoint = "REGIONAL"
+            # to make sure that default is always REGIONAL
+            self.domain["EndpointConfiguration"] = "REGIONAL"
+        elif endpoint not in ["EDGE", "REGIONAL"]:
+            raise InvalidResourceException(
+                self.logical_id,
+                "EndpointConfiguration for Custom Domains must be" " one of {}.".format(["EDGE", "REGIONAL"]),
```

Comment:
nit: is the empty "" needed after `must be`?

Reply:
just Black being annoying. will fix